libs/ui/src/bmd_paper_ballot.dual_language.test.tsx

```diff
@@ -0,0 +1,397 @@
+      (c): c is YesNoContest => c.type === 'yesno'
+    );
+    render(
+      <BmdPaperBallot
+        ballotStyleId={spanishBallotStyle.id}
+        electionDefinition={{
+          ...electionDefinition,
+          election: { ...election, contests: [contest] },
+        }}
+        isLiveMode
+        precinctId={spanishBallotStyle.precincts[0]}
+        votes={{ [contest.id]: [contest.yesOption.id] }}
+        onRendered={_.noop}
+      />
+    );
+
+    await waitFor(() =>
+      expect(getLanguageContext()?.currentLanguageCode).toEqual(
+        LanguageCode.ENGLISH
+      )
+    );
+
+    expectDualLanguageString({
+      key: ElectionStringKey.CONTEST_TITLE,
+      subKey: contest.id,
+    });
+
+    expectDualLanguageString({
+      key: ElectionStringKey.CONTEST_OPTION_LABEL,
+      subKey: contest.yesOption.id,
+    });
+  });
+
+  test('yes/no contest undervote', async () => {
+    const contest = find(
+      contests,
+      (c): c is YesNoContest => c.type === 'yesno'
+    );
+
+    render(
+      <BmdPaperBallot
+        ballotStyleId={spanishBallotStyle.id}
+        electionDefinition={{
+          ...electionDefinition,
+          election: { ...election, contests: [contest] },
+        }}
+        isLiveMode
+        precinctId={spanishBallotStyle.precincts[0]}
+        votes={{}}
+        onRendered={_.noop}
+      />
+    );
+
+    await waitFor(() =>
+      expect(getLanguageContext()?.currentLanguageCode).toEqual(
+        LanguageCode.ENGLISH
+      )
+    );
+
+    expectDualLanguageString({
+      key: ElectionStringKey.CONTEST_TITLE,
+      subKey: contest.id,
+    });
+
+    expectDualLanguageString({ key: 'noteBallotContestNoSelection' });
+  });
+});
+
+describe('English ballot style', () => {
+  const englishBallotStyle = find(
+    election.ballotStyles,
+    (b) => b.languages?.[0] === LanguageCode.ENGLISH
+  );
+
+  const contests = getContests({ ballotStyle: englishBallotStyle, election });
+
+  test('all votes filled in', async () => {
+    const votes: VotesDict = _(contests)
+      .keyBy((c) => c.id)
+      .mapValues((c) =>
+        c.type === 'yesno' ? generateYesNoVote(c) : generateCandidateVotes(c)
+      )
+      .value();
+
+    const { container } = render(
+      <BmdPaperBallot
+        ballotStyleId={englishBallotStyle.id}
+        electionDefinition={electionDefinition}
+        isLiveMode
+        precinctId={englishBallotStyle.precincts[0]}
+        votes={votes}
+        onRendered={_.noop}
+      />
+    );
+
+    await waitFor(() => expect(getLanguageContext()).toBeDefined());
+
+    // Change app-wide language to Spanish to verify ballot language is not
+    // affected:
+    act(() => getLanguageContext()!.setLanguage(LanguageCode.SPANISH));
+    await waitFor(() =>
+      expect(getLanguageContext()?.currentLanguageCode).toEqual(
+        LanguageCode.SPANISH
+      )
+    );
+
+    expect(container).not.toHaveTextContent(
+      new RegExp(`\\[${LanguageCode.SPANISH}\\]`)
+    );
```

Comment:
Since this check is tied to the implementation of `generateMockUiString` above, might be worth putting construction of this `[languageCode]` prefix in a shared function that `generateMockUiString` also uses. Wouldn't want us to change the output of `generateMockUiString` and then have this check still succeed, but for the wrong reason, potentially masking bugs.